### ISSUE TITLE
Hitting delte in empty gene box removes last gene

### ISSUE THREAD

**Issue by signechambers1** [chanzuckerberg/single-cell-data-portal#4156]
@GarabetY found this bug. Steps to reproduce:

1. Prod
2. Add spleen
3. Click into gene drop down, select the first three genes
4. Click in empty gene box
5. Hit delete on your keyboard
6. The last gene you selected is removed from the dot plot

Desired solution: Nothing should happen in this case.

**Comment by seve** [comment]
High probability of SDS bug

**Comment by tihuan** [comment]
Just confirmed that this is SDS thing! Will double check if it's a feature or a bug lol

**Comment by tihuan** [comment]
Just asked SDS support channel for verification if the behavior is a feature or a bug!

https://czi-sci.slack.com/archives/C032S43KKFV/p1678296205703739?thread_ts=1678296154.403999&cid=C032S43KKFV

**Comment by tihuan** [comment]
Masoud from SDS will take a look at this today!

**Comment by seve** [comment]
I'm going to reach out to Masoud about this today.

**Comment by seve** [comment]
The fix will come in the next version of SDS, not sure if we have any blockers preventing a version bump, but I'll investigate this when the update is released.

**Comment by atarashansky** [comment]
We need to update SDS to get this fixed.

**Comment by signechambers1** [comment]
Sure, here it is.

https://user-images.githubusercontent.com/70176538/228310336-10e663d7-eba1-4e93-a38b-05e7ac3cfac1.mov

**Comment by ashin-czi** [comment]
ohhhhhhhhhhhhhh okay I was hitting Delete instead of Backspace

Thank you!! I'll look into it

**Comment by tihuan** [comment]
SDS has fixed the bug, so hopefully updating `czifui` on our repo will fix the bug on our end! Thanks for doing this, @ashin-czi 🙏

**Comment by ashin-czi** [comment]
hmmmm upgraded to czifui v14.4.1 but still seeing the bug, but I know it's updated cause some things look slightly different. It seems fixed on my local story book though

**Comment by tihuan** [comment]
Ah, I think the fix wasn't applied to MenuSelect since I could still repro it on:

https://chanzuckerberg.github.io/sci-components/?path=/story/menuselect-to-be-depreciated--multi-select-with-search

The lowest effort way is just to apply to same fix to MenuSelect, release it, and then we upgrade again

I actually tried using Dropdown when I was doing the migration to MUIv5, but ran into some styling issue. Maybe it's worth a quick try again to see if that's still the case. If it's still hard to upgrade, let's just go with the lowest effort way above

**Comment by ashin-czi** [comment]
@signechambers1 @karenliang0 @hthomas-czi @tihuan 

rdev: https://gene-select-delete-frontend.rdev.single-cell.czi.technology/gene-expression

I updated czifui to v14.4.1

There are some noticeable visual changes, specifically the left sidebar filters. Just wanted to make sure if everyone was okay with this and to see if there were other cases where components look different and need to be altered!

**Comment by tihuan** [comment]
Ooh amazing! Thanks for migrating to `Dropdown` 🤩 👏 ! 

Left is new, right is old:

<img width="1489" alt="Screenshot 2023-03-29 at 6 34 25 AM" src="https://user-images.githubusercontent.com/6309723/228556309-771c6797-ddc3-4ebf-9407-527e2c78547f.png">

**Comment by karenliang0** [comment]
I agree - the header (filters) as well as the filter dropdowns themselves are blending together. Also there's a lot more spacing between each filter dropdown which I don't think is needed? Will look into this today!

**Comment by ashin-czi** [comment]
@hthomas-czi @karenliang0 Any updates on this? Did we want to go with another SDS style?

**Comment by tihuan** [comment]
PR out to fix SDS MenuSelect!

https://github.com/chanzuckerberg/sci-components/pull/413

Once this is merged and released, I'll ping @ashin-czi to use the latest `czifui` and we can go back to use `MenuSelect` for a little longer

Thanks all!

**Comment by tihuan** [comment]
I think SDS is releasing today! Will keep you posted, @ashin-czi 👍 

CC: @cathystoli

**Comment by tihuan** [comment]
Good news! v14.5.0 is just released with the fix ✨ `MenuSelect` should no longer have the bug

@ashin-czi please kindly try upgrading `czifui` and hopefully it'll just work lol

https://chanzuckerberg.github.io/sci-components/?path=/story/menuselect-to-be-depreciated--multi-select-with-search

Thank you![demo.gif](https://images.zenhubusercontent.com/5ebed8780d2d150fc281ad55/734a2999-e2e3-46da-a161-efa8d3b2d411)!

**Comment by GarabetY** [comment]
Amazing! This looks great. Thank you so much!

All the best,
Garabet
